### ISSUE TITLE
Tabular: NeuralNetClassifier does not use GPU

### ISSUE THREAD

**Issue by ghazaryanartashes** [awslabs/autogluon#696]
I am training the tabular classification model. I have installed the GPU version of MXnet and the latest version of autogluon. However, the training of NeuralNetClassifier is very slow and in most cases runs out of time and skips the model. I checked nvidia-smi and I guess that the model just doesn't use GPU.

**Comment by Innixma** [comment]
Can you try using:

`predictor = task.fit(train_data, label=label, ngpus_per_trial=1)`?

GPU usage is not enabled by default, it needs to be specified. This functionality will become more clear in the next release as GPU usage is only partially implemented (Neural network and CatBoost use GPU if specified, but the other models don't)

**Comment by ghazaryanartashes** [comment]
Dear @Innixma thanks for the response,
I was using ngpus_per_trial, but still, the training is very slow and again nvidia-smi doesn't show any GPU usage.

**Comment by Innixma** [comment]
Interesting, thanks for the clarification. I am on vacation next week but will take a look once I return. We are planning to refactor the usage of GPU for v0.1 and I will ensure this is resolved for v0.1.

**Comment by jwmueller** [comment]
@ghazaryanartashes  Did you call fit with `verbosity=4` and check the NN is actually training in the output logs which should print updates every epoch?

The NN should train on GPU if you've properly installed GPU-version of MXNet, but FYI GPU training is not actually that much faster for this NN as it's a fairly small network. The NN model is expected to train slowly if you have high-dimensional data, in which case you can try specifying smaller hyperparameter values for the NN, or simply removing it from the set of models that are trained in `fit()` via the `excluded_model_types` argument.

If you'd like more help, it'll be helpful if can post your code, the output logs from `fit()`, as well as your OS and the output of `pip freeze` to show which package versions you're using.

**Comment by Innixma** [comment]
In v0.1 and pre-release, NeuralNet uses the GPU if specified: #896
